src/asset/index.ts

```diff
@@ -3,43 +3,50 @@ import {getAddress} from "../rpc/getAddress";
 import {executeAssetOperation} from "./action";
 import formatBalance from "@polkadot/util/format/formatBalance";
 import {Balance} from "@polkadot/types/interfaces";
+import {SnapConfig} from "../configuration/interfaces";
+import {getConfiguration} from "../configuration";
 
 const assets: Map<string, Asset> = new Map<string, Asset>();
 
-
 function getIdentifier(origin: string, id: string): string {
   return `${origin}_${id}`;
 }
 
-export function getKusamaAsset(assetId: string, balance: number|string|Balance, address: string): Asset {
+export function getPolkadotAssetDescription(
+  balance: number|string|Balance, address: string, configuration: SnapConfig
+): Asset {
   return {
-    balance: formatBalance(balance, {decimals: 12, withSi: true, withUnit: false}),
-    customViewUrl: `https://polkascan.io/pre/kusama/account/${address}`,
+    balance: formatBalance(balance, {decimals: configuration.unit.decimals, withSi: true, withUnit: false}),
+    customViewUrl: configuration.unit.customViewUrl ||
+        `https://polkascan.io/pre/${configuration.networkName}/account/${address}`,
     decimals: 0,
-    identifier: assetId,
-    image: 'https://img.techpowerup.org/200330/kusama.png',
-    symbol: 'KSM',
+    identifier: configuration.unit.assetId,
+    image: configuration.unit.image || "https://svgur.com/i/Jiy.svg",
```

Comment:
@mpetrunic I actually think this should default to an empty string (_a.k.a. generic metamask image_), what do you think?

Reply:
sure